connectors/sources/sharepoint_online.py

```diff
@@ -1174,97 +1194,121 @@ async def get_access_control(self):
+                    self._logger.debug(f"We've already seen {id}")
+                    return True
 
             return False
 
-        def update_already_seen(email_, username_):
-            # We want to make sure to not add 'None' to the already seen sets
-            if email_:
-                already_seen_emails.add(email_)
+        def update_already_seen(*ids):
+            for id in ids:
+                # We want to make sure to not add 'None' to the already seen sets
+                if id:
+                    already_seen_ids.add(id)
 
-            if username_:
-                already_seen_usernames.add(username_)
+        async def process_user(user):
+            email = user.get("EMail", user.get("mail", None))
+            username = user.get("UserName", user.get("userPrincipalName", None))
+            self._logger.debug(f"Detected a person: {username}: {email}")
 
-        async for site_collection in self.client.site_collections():
-            async for site in self.client.sites(
-                site_collection["siteCollection"]["hostname"],
-                self.configuration["site_collections"],
-            ):
-                self._logger.debug(f"Looking at site: {site['id']}")
-                async for user_information in self.client.user_information_list(
-                    site["id"]
-                ):
-                    user = user_information["fields"]
+            if _already_seen(email, username):
+                return None
 
-                    if is_domain_group(user):
-                        self._logger.debug(
-                            f"Detected a domain group: {user.get('Name')}"
-                        )
-                        domain_group_id = _domain_group_id(user.get("Name"))
-                        self._logger.debug(
-                            f"Detected domain groupId as: {domain_group_id}"
-                        )
+            update_already_seen(email, username)
 
-                        if domain_group_id:
-                            async for member_email, member_username in _emails_and_usernames_of_domain_group(
-                                domain_group_id,
-                                self.client.group_members,
-                            ):
-                                if _already_seen(member_email, member_username):
-                                    continue
+            person_access_control_doc = await self._user_access_control_doc(user)
+            if person_access_control_doc:
+                return person_access_control_doc
 
-                                update_already_seen(member_email, member_username)
+        if not self.configuration["fetch_users_by_site"]:
+            self._logger.info("Fetching all users")
+            async for user in self.client.users():
+                user_doc = await process_user(user)
+                if not user_doc:
+                    continue
+                yield user_doc
```

Comment:
```suggestion
                if user_doc:
                    yield user_doc
```

nit: super small nit